hub/src/server/drivers/AzDriver.ts

```diff
@@ -186,16 +188,32 @@ class AzDriver implements DriverModel, DriverModelTestMethods {
     */
     const maxBuffers = 1
 
-    try {
-      await azure.uploadStreamToBlockBlob(
-        azure.Aborter.none, args.stream,
-        blockBlobURL, bufferSize, maxBuffers, {
-          blobHTTPHeaders: {
-            blobContentType: args.contentType,
-            blobCacheControl: this.cacheControl || undefined
-          }
+    const options: azure.IUploadStreamToBlockBlobOptions = {
+      blobHTTPHeaders: {
+        blobContentType: args.contentType,
+        blobCacheControl: this.cacheControl || undefined
+      },
+      accessConditions: {
+        modifiedAccessConditions: {
+          ifMatch: args.ifMatch,
+          ifNoneMatch: args.ifNoneMatch
         }
+      }
+    }
+
+    try {
+      const uploadResult = await azure.uploadStreamToBlockBlob(
+        azure.Aborter.none, args.stream, blockBlobURL, bufferSize, maxBuffers, options
       )
+
+      // Return success and url to user
+      const readURL = this.getReadURLPrefix()
+      const publicURL = `${readURL}${azBlob}`
+      logger.debug(`Storing ${azBlob} in ${this.bucket}, URL: ${publicURL}`)
+      return {
+        publicURL,
+        etag: uploadResult.eTag.replace(/^"|"$/g, '')
+      }
     } catch (error) {
       logger.error(`failed to store ${azBlob} in ${this.bucket}: ${error}`)
```

Comment:
Should this check for Azure's conditional failure error and throw the same error as the generic version?
e.g.
```js
if (isAzureConditionalFailureResponse(error)) {
  throw new PreconditionFailedError(...)
}
```

Otherwise I think this would result in gaia returning a 500?